darts/models/regression_model.py

```diff
@@ -39,263 +51,226 @@ def __init__(self,
+            raise_log(
+                Exception("Provided model object must have a predict() method", logger)
+            )
 
         self.model = model
-        if self.lags is not None and self.lags_exog is None:
-            self.max_lag = max(self.lags)
-        elif self.lags is None and self.lags_exog is not None:
-            self.max_lag = max(self.lags_exog)
-        else:
-            self.max_lag = max([max(self.lags), max(self.lags_exog)])
-        self._fit_called = False
 
-    def fit(self, series: TimeSeries, exog: Optional[TimeSeries] = None, **kwargs) -> None:
-        """ Fits/trains the model using the provided list of features time series and the target time series.
+        # turning lags into array of int or None
+        self.lags, self.lags_covariates = _process_lags(lags, lags_covariates)
 
-        Parameters
-        ----------
-        series : TimeSeries
-            TimeSeries object containing the target values.
-        exog : TimeSeries, optional
-            TimeSeries object containing the exogenous values.
-        """
-        super().fit(series, exog)
-        raise_if(exog is not None and self.lags_exog is None,
-            "`exog` not None in `fit()` method call, but `lags_exog` is None in constructor. "
-        )
-        raise_if(exog is None and self.lags_exog is not None,
-            "`exog` is None in `fit()` method call, but `lags_exog` is not None in constructor. "
-        )
-        self.exog_columns = exog.columns.values.tolist() if exog is not None else None
-        self.nr_exog = exog.width*len(self.lags_exog) if exog is not None else 0
-
-        # Rename column if necessary
-        target_column = str(series.columns[0])
-        if self.exog_columns is not None and target_column in self.exog_columns:
-            series = TimeSeries.from_dataframe(series.pd_dataframe().rename({target_column: "Target"}))
-        self.target_column = series.columns[0]
-
-        # Prepare data
-        training_x = self._create_training_data(series=series, exog=exog)
-        training_y = series[training_x.time_index]
-
-        # Fit model
-        if "series" in signature(self.model.fit).parameters:
-            self.model.fit(
-                series=training_y,
-                exog=training_x,
-                **kwargs
-            )
+        if self.lags is not None:
+            self.lags_indices = np.array(self.lags) * (-1)
         else:
-            self.model.fit(
-                training_x.pd_dataframe(),
-                training_y.pd_dataframe().values.ravel(),
-                **kwargs
-            )
+            self.lags_indices = None
 
-        if self.max_lag == 0:
-            self.prediction_data = pd.DataFrame(columns=series.stack(other=exog).columns)
-        elif exog is not None:
-            self.prediction_data = series.stack(other=exog)[-self.max_lag:]
+        if self.lags_covariates is not None:
+            if 0 in self.lags_covariates:
```

Comment:
As far as I understood, the regression model was already using the 0 index as "time of prediction" covariate (e.g., comment at line 261, develop branch, regression_model.py), thus I kept the same behavior.

I had to wrap around `SequentialDataset` and `SimpleInferenceDataset` since they did not support "covariates at prediction time". Anyway, I saw that recently some commits in `SimpleInferenceDataset` were going into this direction as well, with the parameter `model_is_recurrent`, thus I recycled some of that logic as well.